lxd/db/instances.go

```diff
@@ -67,15 +56,32 @@ import (
 //go:generate mapper stmt -p db -e instance update struct=Instance
 //
 //go:generate mapper method -p db -e instance List
-//go:generate mapper method -p db -e instance Get
+//go:generate mapper method -p db -e instance List-by-Project
+//go:generate mapper method -p db -e instance List-by-Project-and-Node
+//go:generate mapper method -p db -e instance List-by-Project-and-Name
+//go:generate mapper method -p db -e instance List-by-Node
+//go:generate mapper method -p db -e instance Get-by-Project
+//go:generate mapper method -p db -e instance Get-by-Project-and-Name
 //go:generate mapper method -p db -e instance ID struct=Instance
 //go:generate mapper method -p db -e instance Exists struct=Instance
 //go:generate mapper method -p db -e instance Create struct=Instance
 //go:generate mapper method -p db -e instance ProfilesRef
+//go:generate mapper method -p db -e instance ProfilesRef-by-Project
+//go:generate mapper method -p db -e instance ProfilesRef-by-Project-and-Node
+//go:generate mapper method -p db -e instance ProfilesRef-by-Project-and-Name
+//go:generate mapper method -p db -e instance ProfilesRef-by-Node
 //go:generate mapper method -p db -e instance ConfigRef
+//go:generate mapper method -p db -e instance ConfigRef-by-Project
+//go:generate mapper method -p db -e instance ConfigRef-by-Project-and-Node
+//go:generate mapper method -p db -e instance ConfigRef-by-Project-and-Name
+//go:generate mapper method -p db -e instance ConfigRef-by-Node
 //go:generate mapper method -p db -e instance DevicesRef
+//go:generate mapper method -p db -e instance DevicesRef-by-Project
+//go:generate mapper method -p db -e instance DevicesRef-by-Project-and-Node
+//go:generate mapper method -p db -e instance DevicesRef-by-Project-and-Name
+//go:generate mapper method -p db -e instance DevicesRef-by-Node
 //go:generate mapper method -p db -e instance Rename
-//go:generate mapper method -p db -e instance DeleteOne
+//go:generate mapper method -p db -e instance DeleteOne-by-Project-and-Name
```

Comment:
@masnax would it be possible to include the part of the generator line here "DeleteOne-by-Project-and-Name" in the generated function somehow (like it its comment), so that it is easy to find the actual generated function (which I am struggling with atm).

E.g. 

`// DeleteInstanceByProjectAndName deletes the instance matching the given key parameters. (DeleteOne-by-Project-and-Name)`

This would really speed up reviewing each function! :)

Reply:
Also a separate idea, and if it takes ages don't worry, but I was thinking it would be really useful to be able to see all of the generated function signatures in one file together, as a Go interface, we would then be able to get a top-level view of the proposed/generated user-experience without having to keep flitting around the generator comments and actual function implementations. CC @stgraber

Reply:
Both of those are great ideas and I don't think they would be much work at all :)